### PR TITLE
fix: uncaught PaymentService typings in payment utils

### PR DESCRIPTION
**Pull request by zomars**
## What does this PR do?

Follow up to #15929

Replaced `findFirst` with `findFirstOrThrow` for better error handling in the booking process. Added additional checks for the presence of `PaymentApp` and its properties to handle missing payment services gracefully.

### What changed?
- Replaced `findFirst` with `findFirstOrThrow` for booking
- Added the `PaymentApp` type import
- Restructured `paymentApp` initialization to ensure proper type casting and null handling
- Added checks to ensure `paymentApp`, `paymentApp.lib`, and `paymentApp.lib.PaymentService` exist before proceeding
- Removed redundant booking existence check

### How to test?
1. Trigger a payment process with an existing booking and observe that it behaves as expected
2. Attempt to trigger a payment process with a non-existent booking and verify that an appropriate error is thrown
3. Ensure that the payment process correctly handles cases where the payment service is missing or improperly configured

---

## Mandatory Tasks (DO NOT REMOVE)

- [ ] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [ ] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com). If N/A, write N/A here and check the checkbox.
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my changes generate no new warnings

### CODE REVIEW

**Comment by zomars** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/calcom/cal.com/16107?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#16107** <a href="https://app.graphite.dev/github/pr/calcom/cal.com/16107?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#16106** <a href="https://app.graphite.dev/github/pr/calcom/cal.com/16106?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#16105** <a href="https://app.graphite.dev/github/pr/calcom/cal.com/16105?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#16104** <a href="https://app.graphite.dev/github/pr/calcom/cal.com/16104?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @zomars and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/trpc/server/routers/viewer/payments.tsx [hunk 1]
```diff
@@ -110,9 +107,11 @@ export const paymentsRouter = router({
         throw new TRPCError({ code: "BAD_REQUEST", message: "Invalid payment credential" });
       }
 
-      const paymentApp = await appStore[paymentCredential?.app?.dirName as keyof typeof appStore];
+      const paymentApp = (await appStore[
+        paymentCredential?.app?.dirName as keyof typeof appStore
+      ]?.()) as PaymentApp | null;
```
Comment (joeauyeung) [inline packages/trpc/server/routers/viewer/payments.tsx:112]: Why are we adding the `?.()`? This wasn't there before

Comment (zomars) [reply packages/trpc/server/routers/viewer/payments.tsx:112]: The App Store is an index of functions. We weren't really calling them. Here's another instance: https://github.com/calcom/cal.com/blob/main/packages/trpc/server/routers/loggedInViewer/integrations.handler.ts#L176

Comment (zomars) [reply packages/trpc/server/routers/viewer/payments.tsx:112]: Basically we we're "awaiting" the function and not the promise it was supposed to return.

**Review by joeauyeung — APPROVED** [review]
LGTM! Love the stacked PRs

**Comment by zomars** [comment]
### Merge activity

* **Aug 8, 1:59 PM EDT**: [Graphite](https://app.graphite.dev/github/pr/calcom/cal.com/16107) couldn't merge this PR because it failed for an unknown reason.

**Pull request merged**